scripts/boot_patch.sh

```diff
@@ -231,12 +231,18 @@ if [ -f kernel ]; then
   # After:  [mov w2, #-32768]
   ./magiskboot hexpatch kernel 821B8012 E2FF8F12 && PATCHEDKERNEL=true
 
-  # Force kernel to load rootfs for legacy SAR devices
-  # skip_initramfs -> want_initramfs
-  $SYSTEM_ROOT && ./magiskboot hexpatch kernel \
-  736B69705F696E697472616D667300 \
-  77616E745F696E697472616D667300 \
-  && PATCHEDKERNEL=true
+  # Don't patch on devices that use dynamic partitions, as these devices always
+  # use a ramdisk and patching their kernel may break booting on devices that
+  # did not correctly implement androidboot.force_normal_boot.
+  DYNAMIC_PARTITIONS=$(grep_get_prop ro.boot.dynamic_partitions /default.prop /vendor/build.prop)
```

Comment:
This should be set as an environment variable as `KEEPVERITY, KEEPFORCEENCRYPT, PATCHVBMETAFLAG, RECOVERYMODE, SYSTEM_ROOT`.